test/modules/TestResultCoordinator/IReportGeneratorFactory.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator
+{
+    /// <summary>
+    /// This defines methods for a test result report generator.
+    /// </summary>
+    interface IReportGeneratorFactory
+    {
+        ITestResultReportGenerator Create(
+            string trackingId,
+            ITestResultComparer<TestOperationResult> testResultComparer,
```

Comment:
remove comparer; you can determine what result comparer to use based on result type in reportMetadata